webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/nodes/NodeInfoService.scala

```diff
@@ -875,6 +876,10 @@ trait NodeInfoServiceCached extends NodeInfoService with NamedZioLogger with Cac
   def getNodeInfos(nodeIds: Set[NodeId]): IOResult[Set[NodeInfo]] = withUpToDateCache(s"${nodeIds.size} nodes infos") { cache =>
     cache.filter(x => nodeIds.contains(x._1)).values.map( _._2).toSet.succeed
   }
+
+  def getNodeInfosSeq(nodeIds: Seq[NodeId]): IOResult[Seq[NodeInfo]] = withUpToDateCache(s"${nodeIds.size} nodes infos") { cache =>
+    nodeIds.map(id => cache(id)).map(_._2).succeed
```

Comment:
it could fail if some requested node id are not in cache (for some inconsistency from somewhere). 
What is the correct thing to do on missing nodeId ? Fail everything or ignore ? 
If fail, something like:
```withUpToDateCache(s"${nodeIds.size} nodes infos") { cache =>
ZIO.foreach(nodeIds) { id => cache.get(id).map(_._2).notOptionnal(s"node with id ${id.value} not in cache, likely an error")} }
```
If ignore:
```withUpToDateCache(s"${nodeIds.size} nodes infos") { cache =>
nodeIds.map(id => cache.get(id).map(_._2)).collect { case Some(info) => info }.succeed
}```

Reply:
ha, very good point. 
to be consistent, we should ignore, so the second approach is the right one. Thank you !